src/server/game/Entities/Unit/Unit.cpp

```diff
@@ -1316,13 +1316,15 @@ void Unit::CalculateMeleeDamage(Unit* victim, CalcDamageInfo* damageInfo, Weapon
             damageInfo->HitInfo     |= HITINFO_GLANCING;
             damageInfo->TargetState  = VICTIMSTATE_HIT;
             int32 leveldif = int32(victim->GetLevel()) - int32(GetLevel());
+            if (leveldif == 0)
+                leveldif = 1;
             if (leveldif > 3)
                 leveldif = 3;
 
             // against boss-level targets - 24% chance of 25% average damage reduction (damage reduction range : 20-30%)
             // against level 82 elites - 18% chance of 15% average damage reduction (damage reduction range : 10-20%)
             int32 const reductionMax = leveldif * 10;
-            int32 const reductionMin = reductionMax - 10;
+            int32 const reductionMin = std::min(1, reductionMax - 10);
```

Comment:
does this code handle a negativ leveldif ?

Reply:
a glancing can only occur against same or higher level, can't be negative.

Reply:
Well but before your change, in case of same level reductionMax was 0 and reductionMin was -10.

Btw is that std::min correct ? With a leveldif of 3, before reductionMin was 20, now it's 1

Reply:
Before my changes, glancing didn't happened against same level. So it was fine.

You're totally right, that's a mistake.
The goal was to obtain a reduction range between 1% and 10% when leveldif is 1. So it's the max of 1 and 0 not the min…

```suggestion
            int32 const reductionMin = std::max(1, reductionMax - 10);
```

Reply:
@Metalaka remember when I asked "does this code handle a negativ leveldif ?" ?

https://gist.github.com/Aokromes/7e65ac8c8d3ed269eb9aef0214941603

>reductionMax = -70
reductionMin = 413603904

Random.cpp:37 in irand ASSERTION FAILED

It's better to handle in the code all the cases so crashes don't happen

Reply:
This was not intended. That negative reduction allow to silently do more damage with glancings that critics.
A level 80 vs a vanilla-boss (lvl63) won an insane 2.75 modifier on glancings.
It's fixed in https://github.com/TrinityCore/TrinityCore/pull/26120

Feel free to add some checks on `leveldif` if you want, but it contiue to seem overkill to me.